File: Best Time to Buy and Sell Stock II/Best_Time_to_Buy_and_Sell_Stock_II.cpp
// 累计每一个递增序列
// Runtime: 4 ms, faster than 99.43% of C++ online submissions for Best Time to Buy and Sell Stock II.
// Memory Usage: 1.6 MB, less than 93.04% of C++ online submissions for Best Time to Buy and Sell Stock II.

class Solution 
{
public:
    int maxProfit(vector<int>& prices) 
    {
        int res = 0;
        for (int i = 1; i < prices.size(); i++)
        {
            if (prices[i] > prices[i - 1])
                res += (prices[i] - prices[i - 1]);
        }
        return res;
    }
};

File: Best Time to Buy and Sell Stock II/Best_Time_to_Buy_and_Sell_Stock_II.py
# Runtime: 40 ms, faster than 97.81% of Python3 online submissions for Best Time to Buy and Sell Stock II.
# Memory Usage: 7.2 MB, less than 89.70% of Python3 online submissions for Best Time to Buy and Sell Stock II.

class Solution:
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        return sum([prices[i] - prices[i - 1] for i in range(1, len(prices)) if prices[i] > prices[i - 1]])
        